Client/Frontend/Browser/Favorites/FavoritesViewController.swift

```diff
@@ -0,0 +1,439 @@
+        
+        do {
+            try frc.performFetch()
+        } catch {
+            log.error("Favorites fetch error: \(String(describing: error))")
+        }
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.addSubview(backgroundView)
+        view.addSubview(collectionView)
+        
+        backgroundView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        collectionView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        collectionView.alwaysBounceVertical = true
+        collectionView.contentInset = UIEdgeInsets(top: 24, left: 0, bottom: 0, right: 0)
+        collectionView.backgroundColor = .clear
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.dragDelegate = self
+        collectionView.dropDelegate = self
+        collectionView.dragInteractionEnabled = true
+        collectionView.keyboardDismissMode = .interactive
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        calculateAppropriateGrid()
+        
+        if let state = KeyboardHelper.defaultHelper.currentState {
+            updateKeyboardInset(state, animated: false)
+        }
+    }
+    
+    func applyTheme(_ theme: Theme) {
+        let blurStyle: UIBlurEffect.Style = theme.isDark ? .dark : .extraLight
+        backgroundView.effect = UIBlurEffect(style: blurStyle)
+        backgroundView.contentView.backgroundColor = theme.colors.home.withAlphaComponent(0.5)
+        collectionView.reloadSections(IndexSet(integer: 0))
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        
+        collectionView.contentInset = collectionView.contentInset.with {
+            $0.left = self.view.readableContentGuide.layoutFrame.minX
+            $0.right = self.view.readableContentGuide.layoutFrame.minX
+        }
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        calculateAppropriateGrid()
+    }
+    
+    private func calculateAppropriateGrid() {
+        let width = collectionView.bounds.width -
+            (layout.sectionInset.left + layout.sectionInset.right) -
+            (collectionView.contentInset.left + collectionView.contentInset.right)
+        // Want to fit _at least_ 4 on all devices, but on larger devices
+        // allowing the cells to be a bit bigger
+        let minimumNumberOfColumns = 4
+        let minWidth = floor(width / CGFloat(minimumNumberOfColumns))
+        // Default width should be 82, but may get smaller or bigger
+        var itemSize = CGSize(width: 82, height: FavoriteCell.height(forWidth: 82))
```

Comment:
Where is this 82 value from?

Reply:
No where really, just a value that was close to original size after slightly altered `FavoriteCell` + CV layout.